### ISSUE TITLE
[Vue 2] Support for vdom in values

### ISSUE THREAD

**Issue by Frizi** [learningequality/vue-intl#5]
Currently there is no way to inject vdom into the interpolation, including components. React-intl support that pretty nicely, as you can just use vdom directly in the props. In case of Vue2, a possible approach would be to use a component with the optional slot for each interpolation parameter.

This example, based on https://github.com/yahoo/react-intl#example
```javascript
data() {
  return {
    name       : 'Eric',
    unreadCount: 1000,
  }
}
```
```html
<formatted-message
  id="welcome"
  default-message="Hello {name}, you have {unreadCount, number} {unreadCount, plural, one {message} other {messages}}"
  :values="{unreadCount}"
>
  <b slot="name">{{ userName }}</b>
</formatted-message>
```

Would output
```html
Hello <b>Eric</b>, you have 1,000 messages.
```

Implementation would be possible via `:is` property. You can specify a component definition object directly, and render function can be generated on the fly based on the interpolation.
https://vuejs.org/v2/guide/components.html#Dynamic-Components

What do you think?

**Comment by rtibbles** [comment]
Yes, I had planned to do a parallel component implementation for vue-intl (as in react-intl), but have yet to get around with it. Will be publishing the Vue2 compatible version shortly, hopefully that can serve as a platform for getting started on this. Would you have time to open a PR for this at all?

**Comment by Frizi** [comment]
I actually am running it in the project for a while, vue2 based. Everything seems to be working just fine. I have implemented this `FormattedMessage` at the beginning and we are using it everywhere. It's a functional component which inspects provided slots, calls `$formatMessage` and produces a vdom tree out of them. I might adapt it for open source soon and make a PR :)

**Comment by rtibbles** [comment]
Awesome! I had been planning to make components, a la React-intl, but had not had time yet. Would be very happy to get a PR with any implementations.

**Comment by Frizi** [comment]
Issue addressed in #9, please continue there
